packages/@react-aria/tabs/src/useTabPanel.ts

```diff
@@ -29,35 +29,10 @@ export interface TabPanelAria {
  * the contents of a tab, and is shown when the tab is selected.
  */
 export function useTabPanel<T>(props: AriaTabPanelProps, state: TabListState<T>, ref: RefObject<Element>): TabPanelAria {
-  let [tabIndex, setTabIndex] = useState(0);
-
   // The tabpanel should have tabIndex=0 when there are no tabbable elements within it.
   // Otherwise, tabbing from the focused tab should go directly to the first tabbable element
   // within the tabpanel.
-  useLayoutEffect(() => {
-    if (ref?.current) {
-      let update = () => {
-        // Detect if there are any tabbable elements and update the tabIndex accordingly.
-        let walker = getFocusableTreeWalker(ref.current, {tabbable: true});
-        setTabIndex(walker.nextNode() ? undefined : 0);
-      };
-
-      update();
-
-      // Update when new elements are inserted, or the tabIndex/disabled attribute updates.
-      let observer = new MutationObserver(update);
-      observer.observe(ref.current, {
-        subtree: true,
-        childList: true,
-        attributes: true,
-        attributeFilter: ['tabIndex', 'disabled']
-      });
-
-      return () => {
-        observer.disconnect();
-      };
-    }
-  }, [ref]);
```

Comment:
@devongovett hi! I've noticed that after extraction of this layoutEffect to `useHasTabbableChild` it does not have a dependencies array anymore, which makes it run on every re-render. I am wondering if that was done on purpose or it was overlooked?

Reply:
I think the comment explains it pretty well. We want synchronous updates when React re-renders, but need to detect changes when child components re-render too. That's where the MutationObserver comes in. https://github.com/adobe/react-spectrum/blob/73414999fcc6afab5c4bc8ec4740fe3dc1eb8502/packages/%40react-aria/focus/src/useHasTabbableChild.ts#L56